### PR TITLE
add test_2d_dambreak_python case with pybind

### PR DESCRIPTION
**Pull request by maiyetum95**

### CODE REVIEW

**Comment by Xiangyu-Hu** [comment]
> _ No description provided. _

Please provide a description to the changes.

File: tests/2d_examples/test_2d_dambreak_python/pybind_tool/pybind_test.py
Comment (maiyetum95) [inline tests/2d_examples/test_2d_dambreak_python/pybind_tool/pybind_test.py:1]: Add the path of the generated files from cpp (shared object in LInux and dynamic link libraries in Windows) to the python script, which should be corresponds to the folder where the file is generated.

File: tests/2d_examples/test_2d_dambreak_python/CMakeLists.txt
Comment (maiyetum95) [inline tests/2d_examples/test_2d_dambreak_python/CMakeLists.txt:1]: If there are several python version in the computer, make sure that the python version here running the python script should be the same with the version which generated dynamic files.

File: .github/workflows/ci.yml
Comment (maiyetum95) [inline .github/workflows/ci.yml:1]: Considering that the pybind is used as a submodule, we should initialize and update the submodule.

File: CMakeLists.txt
Comment (maiyetum95) [inline CMakeLists.txt:1]: In the main CMakeLists.txt, we add Position Independent Code (PIC) for Linux compile. 
Pybind only have head files, so we use add_subdirectory to add the module.

File: SPHINXsys/src/for_2D_build/geometries/level_set_supplementary.cpp
Comment (maiyetum95) [inline SPHINXsys/src/for_2D_build/geometries/level_set_supplementary.cpp:1]: Fix the problem when use Linux to compile the shared objects.

File: tests/2d_examples/test_2d_dambreak_python/pybind_tool/pybind_tool.py
Comment (maiyetum95) [inline tests/2d_examples/test_2d_dambreak_python/pybind_tool/pybind_tool.py:1]: use python to run the 2d dambreak case, there are two arguments can be changed. The first is for setting restart step, the second is the total running time for the case.

File: .github/workflows/ci.yml
Comment (maiyetum95) [reply .github/workflows/ci.yml:1]: This pybind11_dir is not necessary if use add subdirectory in cmakelists

File: tests/2d_examples/test_2d_dambreak_python/dambreak_python.cpp
Comment (maiyetum95) [inline tests/2d_examples/test_2d_dambreak_python/dambreak_python.cpp:1]: didn't change the main part when compared with the original dambreak case, only add cmakeTest() function for ctest when call the pybind_test.py

**Comment by Xiangyu-Hu** [comment]
Its test finished in 0.11 second? It is strange.

File: tests/2d_examples/test_2d_dambreak_python/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+     DESTINATION ${BUILD_BIND_PATH})
+
+aux_source_directory(. DIR_SRCS)
+pybind11_add_module(${PROJECT_NAME} ${EXECUTABLE_OUTPUT_PATH} ${DIR_SRCS} )
+set_target_properties(${PROJECT_NAME} PROPERTIES VS_DEBUGGER_WORKING_DIRECTORY "${EXECUTABLE_OUTPUT_PATH}")
+target_link_libraries(${PROJECT_NAME} PRIVATE sphinxsys_2d)
+
+if(${CMAKE_SYSTEM_NAME} MATCHES Windows)
+    set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
+    set(PYTHON_EXECUTABLE "C:/hostedtoolcache/windows/Python/3.9.13/x64/python3.exe")
```
Comment (FabienPean-Virtonomy) [inline tests/2d_examples/test_2d_dambreak_python/CMakeLists.txt:29]: This is set automatically when the Python dependency is correctly in the CMake files, with `find_package(Python ...)`, see https://cmake.org/cmake/help/latest/module/FindPython.html

**Review by FabienPean-Virtonomy — CHANGES_REQUESTED** [review]
Please write a description and reasons to the PR by editing your main comment in the conversation.

**Pull request closed without merging**